config/config-api/src/com/thoughtworks/go/config/ServerConfig.java

```diff
@@ -37,6 +37,7 @@
     @ConfigAttribute(value = "purgeUpto", optional = true, allowNull = true) private Double purgeUpto;
     @ConfigAttribute(value = "jobTimeout", optional = true) private String jobTimeout = "0";
     @ConfigAttribute(value="agentAutoRegisterKey", optional = true, allowNull = true) private String agentAutoRegisterKey;
+    @ConfigAttribute(value="webhookSecret", optional = true, allowNull = true) private String webhookSecret;
```

Comment:
The `ensureWebhookSecretExists` will be invoked on a config save everytime, so although it's marked `optional`, a value will be generated even if intentionally left blank.